src/tools/cleaner.js

```diff
@@ -43,11 +43,35 @@ const recurseReplaceBank = (obj, bank = {}) => {
     if (typeof out !== 'string') {
       return out;
     }
+
     Object.keys(bank).forEach(key => {
       // Escape characters (ex. {{foo}} => \\{\\{foo\\}\\} )
-      const s = String(key).replace(/[-[\]/{}()\\*+?.^$|]/g, '\\$&');
-      const re = new RegExp(s, 'g');
-      out = out.replace(re, bank[key]);
+      const escapedKey = String(key).replace(/[-[\]/{}()\\*+?.^$|]/g, '\\$&');
+      const matchesKey = new RegExp(escapedKey, 'g');
+
+      if (!matchesKey.test(out)) {
+        return;
+      }
+
+      const valueParts = out.split(/({{.*?}})/).filter(Boolean);
+      const replacementValue = bank[key];
+      const isPartOfString = valueParts.length > 1;
+
+      if (isPartOfString) {
+        if (_.isArray(replacementValue) || _.isPlainObject(replacementValue)) {
+          throw new TypeError(
+            [
+              'Cannot interpolate objects or arrays into a string.',
+              `You've sent an ${getObjectType(replacementValue)},`,
+              `which will get coerced to ${replacementValue}`
+            ].join(' ')
```

Comment:
is there a reason this is an array vs a string that prettier breaks up automatically?

Reply:
Yeah, I just found it easier to read this way than the way that prettier was formatting it. I don't feel strongly about, though. Can totally be just a string.